components/admin_console/filter/team_filter_dropdown/team_filter_dropdown.tsx

```diff
@@ -152,14 +157,24 @@ class TeamFilterDropdown extends React.PureComponent<Props, State> {
             const {data} = response;
             searchResults = page > 0 ? this.state.searchResults.concat(data.teams) : data.teams;
             searchTotal = data.total_count;
+            this.setState({page, loading: false, searchResults, searchTotal, searchTerm: term});
+            return;
         }
-        this.setState({page, loading: false, searchResults, searchTotal, searchTerm: term});
+        this.searchRetryInterval = this.searchRetryInterval * 2;
+        this.searchRetryId = window.setTimeout(this.searchTeams.bind(null, term, page), this.searchRetryInterval)
```

Comment:
Hey @mkraft @hahmadia I added this search retry logic here which retries search if it fails (after 600ms then 1200ms then 2400ms doubling every time). After discussing with Michael this was the solution that he wanted to be implemented instead of having the user click on a retry button it would keep retrying at longer intervals and show a loading spinner in the meantime... If the search term changes then the search retry is cleared